lib/wibox/widget/textbox.lua

```diff
@@ -73,13 +73,10 @@ end
 -- @treturn number The preferred width.
 -- @treturn number The preferred height.
 function textbox:get_preferred_size(s)
-    local dpi
-    if s then
-        dpi = screen[s].dpi
-    else
+    if not s then
         gdebug.deprecate("textbox:get_preferred_size() requires a screen argument", {deprecated_in=5, raw=true})
-        dpi = beautiful.xresources.get_dpi()
     end
+    local dpi = beautiful.screen_font_dpi(s)
```

Comment:
This changes the semantics of which DPI value is used when this function is called with `s=nil`. Previously, it used some concept of "core DPI" while now it uses the DPI of the primary screen.
@Elv13 Does that count as a regression?